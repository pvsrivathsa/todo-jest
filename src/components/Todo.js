import React, { PropTypes } from 'react'

export default class Todo extends React.Component {
  render() {
    let { completed, text, onClick } = this.props
    return (
      <li
        onClick={onClick}
        style={{
          textDecoration: completed ? 'line-through' : 'none'
        }}
      >
        {text}
      </li>
    )
  }
}

Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired
}
