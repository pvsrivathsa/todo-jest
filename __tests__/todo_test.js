import React from 'react';
import Todo from '../src/components/Todo';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

/* Snapshot */
test('should render the todo component', () => {
	const component = renderer.create(
		<Todo text="Attend Chennai JS Meetup" completed={false} onClick={jest.fn()}/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})

/* Shallow */
test('should render the given todo title', () => {
	const todoTitle = "todo_title";
	const todo = shallow(<Todo text={todoTitle} completed={false} onClick={jest.fn()}/>);
	expect(todo.text()).toBe(todoTitle);
})

test('should call onClick', () => {
	let mockOnClick = jest.fn();
	const todo = shallow(<Todo text="todo_title" completed={false} onClick={mockOnClick}/>);

	todo.find('li').simulate('click');
	expect(mockOnClick.mock.calls.length).toBe(1);
})

test('should mark as completed', () => {
	const todo = shallow(<Todo text="todo_title" completed={true} onClick={jest.fn()}/>);

	let li = todo.find('li');
	expect(li.prop('style').textDecoration).toBe('line-through');
})



