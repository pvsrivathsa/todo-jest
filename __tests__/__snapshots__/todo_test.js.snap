exports[`test should render the todo component 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }>
  Attend Chennai JS Meetup
</li>
`;
